rfcs/0000-dns-cluster-aware-delegation.md

```diff
@@ -0,0 +1,160 @@
+
+Making the controller aware of other clusters is done via specially labeled secret resources containing kubeconfig data that exist in the `kuadrant-system` namespace.
+
+The kubeconfig data should be generated from an appropriately scoped service account that exists on the target cluster.
+
+When a cluster secret is created, the DNS controller on that cluster will automatically detect it and start watching for DNSRecord resources on that external cluster as well as its own.
+
+**Example** Secret containing kubeconfig data
+```yaml
+apiVersion: v1
+kind: Secret
+data:
+  kubeconfig: <kubeconfig data>
+metadata:
+  labels:
+    kuadrant.io/multicluster-kubeconfig: "true"
+  name: kind-kuadrant-local-2
+  namespace: kuadrant-system
+type: Opaque
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## CRD Provider
+
+A new provider implementation that allows a DNSRecord resource to act as a zone record will be created. 
+Endpoints from the source are injected into the target "zone" record using the same plan logic as all other providers. 
+A zone record could be getting updated by many DNSRecords just like any other providers zone.
+
+## Authority Delegation
+
+A new API on the DNSRecord that allows a DNSRecord (and DNSPolicy) to signal that it is delegating authority of publishing to another DNSRecord. 
+Internally uses the CRD provider described above to do this. 
+Assigns a role to the delegation, one of "primary" or "remote" to signal what this particular records role is. 
+Primary records will update/create the authoritative record on the same cluster and supply the provider credentials, remote records only update existing zone records.
+
+## Multicluster
+
+Updates the dns operator to allow watches for DNSRecords on different clusters, ultimately allowing a single controller to reconcile DNSRecords from multiple clusters. 
+Clusters are configured via Secrets containing kubeconfig data and labelled as cluster secrets.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* Adds another path through the code that isn't strictly necessary for the majority of providers.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The initial idea here was to look at a way to "push" records from multiple "remote" clusters to a central "primary" cluster that would be in charge of combining those records into a single "authoritative" record and ultimately in charge of publishing that record to the provider (CoreDNS). 
+This would require all clusters to have credentials capable of writing to the "primary" cluster.
+
+However, it appears to be a more common pattern in kubernetes projects that deal with multi cluster communication to have the "primary" clusters read from multiple "remote" clusters. 
+This would also only require the "primary" to have read access, or very limited write (status only) access to each cluster.
```

Comment:
I see it is called out here on the perms. But I think we should put it under the reference level stuff